hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -664,12 +689,12 @@ private ByteString getCachedStateMachineData(Long logIndex, long term,
             future.complete(
                 getCachedStateMachineData(entry.getIndex(), entry.getTerm(),
                     requestProto));
-          } catch (IOException e) {
+          } catch (Exception e) {
             metrics.incNumReadStateMachineFails();
             future.completeExceptionally(e);
           }
           return future;
-        }, getChunkExecutor(requestProto.getWriteChunk()));
+        }, getReadChunkExecutor(requestProto.getWriteChunk()));
```

Comment:
Instead of introducing another set of executors for read chunk operation, can the cache lookup be executed in the per-container executor returned by `getCommandExecutor`?

Reply:
@adoroszlai  Thank you very much. I will try it.

Reply:
I think, lookUp from the cache can be done in the same thread where readStateMachineData is called. I don't think there is a need here to delegate the cache lookup to a different thread. Reading and writing from the disk can be done using the same set of chunk executor threads.

Reply:
@adoroszlai  How do you think of @bshashikant 's advice ? I think it's also reasonable, because lookup from cache is so fast, there is no need to do it in another thread.

Reply:
Sounds good.